src/main/java/com/plotsquared/holoplots/HoloPlots.java

```diff
@@ -0,0 +1,76 @@
+package com.plotsquared.holoplots;
+
+import com.plotsquared.core.configuration.caption.Caption;
+import com.plotsquared.core.plot.Plot;
+import com.plotsquared.holoplots.provider.HologramProvider;
+import net.kyori.adventure.text.Component;
+import org.bukkit.inventory.ItemStack;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+public interface HoloPlots {
+
+    /**
+     * Logger to use for logging purposes across {@link HologramProvider HologramProviders} etc.
+     *
+     * @return The logger
+     */
+    @NonNull Logger logger();
+
+    /**
+     * The currently used provider based on the other available plugins
+     *
+     * @return {@link HologramProvider} implementation
+     */
+    @NonNull HologramProvider provider();
+
+    /**
+     * Packs a {@link Plot} kinda-unique into a long by putting the {@link com.plotsquared.core.plot.PlotArea
+     * PlotAreas} hash code into the first 4 bytes of the long and the {@link com.plotsquared.core.plot.PlotId PlotIds} hash
+     * code into the following / last 4 bytes.
+     *
+     * @param plot The plot to hash "uniquely" across multiple worlds / {@link com.plotsquared.core.plot.PlotArea PlotAreas}
+     * @return The hash of the plot
+     */
+    long hashPlot(@NonNull Plot plot);
```

Comment:
What happens on collisions?

Reply:
Well. If using HolographicDisplays a hologram could be overridden, where the old plot looses its hologram and the new plot gets theirs.
And for unlinking, the pending task would be cancelled, like: You unlink a plot, it'll be stored in a map by its hash and once an unlink finished it'll be fetched from that map. If a plot with the same hash is unlinked between the hologram from the first one wouldn't be updated. 

Either way, it should be quite an edge case I think. I could create an Object containing the plot area id and plot id - my intention was to reduce the memory overhead, especially for storing the HolographicDisplays holograms

Reply:
I think the (rare) cases where this might cause headache aren't worth the reduced memory overhead

Reply:
Fair. The Plot instance does exist either way - so I guess I can just use its reference